File: kotlin-ide/j2k/new/testData/newJ2k/annotations/jetbrainsNotNull.java
// !FORCE_NOT_NULL_TYPES: false
// !SPECIFY_LOCAL_VARIABLE_TYPE_BY_DEFAULT: true
package test;

import org.jetbrains.annotations.NotNull;

public class Test {
  @NotNull String myStr = "String2";

  public Test(@NotNull String str) {
    myStr = str;
  }

  public void sout(@NotNull String str) {
    // UNNECESSARY_NOT_NULL_ASSERTION heuristic does not work any more, instead we can skip generating !! altogether
    System.out.println(str);
  }

  @NotNull
  public String dummy(@NotNull String str) {
    return str;
  }

  public void test() {
    sout("String");
    @NotNull String test = "String2";
    sout(test);
    sout(dummy(test));

    new Test(test);
  }
}

File: kotlin-ide/j2k/new/testData/newJ2k/annotations/jetbrainsNotNull.kt
// !FORCE_NOT_NULL_TYPES: false
// !SPECIFY_LOCAL_VARIABLE_TYPE_BY_DEFAULT: true
package test

class Test(str: String) {
    var myStr = "String2"
    fun sout(str: String) {
        // UNNECESSARY_NOT_NULL_ASSERTION heuristic does not work any more, instead we can skip generating !! altogether
        println(str)
    }

    fun dummy(str: String): String {
        return str
    }

    fun test() {
        sout("String")
        val test = "String2"
        sout(test)
        sout(dummy(test))
        Test(test)
    }

    init {
        myStr = str
    }
}